internal/controller/fluxinstance_controller.go

```diff
@@ -359,12 +359,12 @@ func (r *FluxInstanceReconciler) build(ctx context.Context,
 		options.Patches += builder.GetProfileMultitenant(obj.GetCluster().TenantDefaultServiceAccount)
 	}
 
-	if err := options.ValidateAndApplyWorkloadIdentityConfig(obj.GetCluster()); err != nil {
+	if err := options.ValidateAndPatchComponents(); err != nil {
 		return nil, err
 	}
 
-	if options.HasNotificationController() {
-		options.Patches += builder.GetProfileNotification(options.Namespace)
+	if err := options.ValidateAndApplyWorkloadIdentityConfig(obj.GetCluster()); err != nil {
```

Comment:
If we are changing the order of these two `options.ValidateAnd...()` calls here, we also need to change in the CLI